relay-general/src/protocol/contexts/trace.rs

```diff
@@ -102,6 +102,10 @@ pub struct TraceContext {
     /// should not ever send this value.
     pub client_sample_rate: Annotated<f64>,
 
+    /// The origin of the trace indicates what created the trace (see [OriginType] docs).
+    #[metastructure(max_chars = "enumlike", allow_chars = "a-zA-Z0-9_.")]
+    pub origin: Annotated<OriginType>,
```

Comment:
@iker-barriocanal, can I somehow add the metastructure to the definition of `OriginType` so we reuse it? I'm new to Rust and Relay and have no clue how to achieve this.